### PR TITLE
Upgrade preact-i18n to 2.3.1-preactx

### LINKED ISSUE #1132: Upgrade preact-i18n
In a  few places, we're accessing translations via the context, rather than via the library's APIs (notably `<Text />`, `withText` and `useText`).

In order to upgrade the library, we either need to temporarily adjust the codebase to use the new context or, alternatively, update instances to use the approved API.

### PR DESCRIPTION
**Pull request by rob-mccann**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

preact-i18n uses the new Context API and has removed support for the legacy API.

I've added a sort of shim in for now into these functions to allow us to rewrite them to use the new API one by one. 

Allows #1128 to be re-added.

Closes #1132

### CODE REVIEW

File: front/src/utils/withIntlAsProp.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { IntlContext } from 'preact-i18n';
+
+const withIntlAsProp = WrappedComponent => props => (
+  <IntlContext.Consumer>{({ intl }) => <WrappedComponent {...{ ...props, intl }} />}</IntlContext.Consumer>
```
Comment (atrovato) [inline front/src/utils/withIntlAsProp.js]: With `{...{ ...props, intl }}`, you copy twice the object.
I think `<WrappedComponent {...props} intl={intl} />` is more efficient, (or `<WrappedComponent {...props, intl} />` only once).

**Review by atrovato — COMMENTED** [review]
That's a good point!!!
All 'context' usages look changed.
I need to test it locally new.

**Review by atrovato — DISMISSED** [review]
Works fine locally.

**Review by Pierre-Gilles — APPROVED** [review]
Looks good to me! I tested on most screens. Couldn't test some integrations (Zigbee2mqtt), l'll merge this on master, build a docker dev tag and ask the community to make sure everything is working properly.

Thank you for your contribution! 👏

**Pull request merged**